### PR TITLE
215 image inpainting

### PR DESCRIPTION
**Pull request by adrianboguszewski**
This is a replacement for #471

### CODE REVIEW

File: notebooks/215-image-inpainting/215-image-inpainting.ipynb [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+   "outputs": [],
+   "source": [
+    "core = Core()\n",
+    "\n",
+    "# Read the model.xml and weights file\n",
+    "model = core.read_model(model=ir_path)\n",
+    "# Load the model on to the CPU\n",
+    "compiled_model = core.compile_model(model=model, device_name=\"CPU\")\n",
+    "# Store the input node names as a list because this model has 2 inputs\n",
+    "input_layers = list(compiled_model.inputs)\n",
```
Comment (Debskij) [inline notebooks/215-image-inpainting/215-image-inpainting.ipynb:126]:
```suggestion
    "input_layer = compiled_model.input(0)\n",
```

File: notebooks/215-image-inpainting/215-image-inpainting.ipynb [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2606632b-41a3-41e4-88ae-935029ec87df",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "N, H, W, C = input_layers[0].shape"
```
Comment (Debskij) [inline notebooks/215-image-inpainting/215-image-inpainting.ipynb:147]:
```suggestion
    "N, H, W, C = input_layer.shape"
```

File: notebooks/215-image-inpainting/215-image-inpainting.ipynb [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "core = Core()\n",
+    "\n",
+    "# Read the model.xml and weights file\n",
+    "model = core.read_model(model=ir_path)\n",
+    "# Load the model on to the CPU\n",
+    "compiled_model = core.compile_model(model=model, device_name=\"CPU\")\n",
+    "# Store the input node names as a list because this model has 2 inputs\n",
```
Comment (Debskij) [inline notebooks/215-image-inpainting/215-image-inpainting.ipynb:125]: We are not storing node names in list below

File: notebooks/215-image-inpainting/215-image-inpainting.ipynb [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+   "metadata": {},
+   "source": [
+    "### Load the model\n",
+    "\n",
+    "Now we will load the IR model. Then:\n",
+    "\n",
+    " 1. Initialize OpenVINO Runtime (Core)\n",
+    " 2. Read the network from *.bin and *.xml files (weights and architecture)\n",
+    " 3. Compile the model for the \"CPU\"\n",
+    " 4. Get input and output names of nodes.\n",
```
Comment (Debskij) [inline notebooks/215-image-inpainting/215-image-inpainting.ipynb:107]: We are not exactly storing output and input names

File: notebooks/215-image-inpainting/215-image-inpainting.ipynb [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+   "source": [
+    "def create_mask(image_width, image_height, size_x=30, size_y=30, number=1):\n",
+    "    \"\"\"\n",
+    "    Create a square mask of defined size on a random location\n",
+    "\n",
+    "    :param: image_width: width of the image\n",
+    "    :param: image_height: height of the image\n",
+    "    :param: size: size in pixels of one side\n",
+    "    :returns:\n",
+    "            mask: grayscale float32 mask of size [image_height,image_width,1]\n",
```
Comment (Debskij) [inline notebooks/215-image-inpainting/215-image-inpainting.ipynb:175]:
```suggestion
    "            mask: grayscale float32 mask of size shaped [image_height, image_width, 1]\n",
```

**Review by Debskij — CHANGES_REQUESTED** [review]

File: notebooks/215-image-inpainting/215-image-inpainting.ipynb [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+   "id": "89ce844f-35c0-4594-a4d4-c8fb1946c7f2",
+   "metadata": {
+    "pycharm": {
+     "is_executing": true
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Download an image\n",
+    "url = \"https://user-images.githubusercontent.com/4547501/165946837-007d9f76-5418-41f3-92c9-2f37091b72dd.png\"\n",
```
Comment (ryanloney) [inline notebooks/215-image-inpainting/215-image-inpainting.ipynb:226]:
```suggestion
    "url = \"https://www.intel.com/content/dam/www/central-libraries/us/en/images/arc-home-hero-128.png.rendition.intel.web.480.360.png\"\n",
```
Please change to something Intel related. If you don't like this image, please suggest an alternative

File: notebooks/215-image-inpainting/215-image-inpainting.ipynb [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+    "image_file = utils.download_file(\n",
+    "    url, filename=\"waterfall.png\", directory=\"data\", show_progress=False, silent=True, timeout=30\n",
```
Comment (ryanloney) [inline notebooks/215-image-inpainting/215-image-inpainting.ipynb:228]:
```suggestion
    "    url, filename=\"laptop.png\", directory=\"data\", show_progress=False, silent=True, timeout=30\n",
```

File: notebooks/215-image-inpainting/215-image-inpainting.ipynb [hunk 8]
```diff
@@ -0,0 +1,350 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "c9c2cbeb-fe8b-4e09-9432-4f4f4bfddc9c",
+   "metadata": {},
+   "source": [
+    "## Image In-painting using OpenVINO\n",
```
Comment (ryanloney) [inline notebooks/215-image-inpainting/215-image-inpainting.ipynb:8]:
```suggestion
    "## Image In-painting with OpenVINO™\n",
```

**Review by ryanloney — COMMENTED** [review]
Please use an Intel-related image for the notebook. I provided a suggestion. 

Also, it would be nice to save the repaired image after applying in-painting as a .png. Can you add a step for it? Otherwise this looks good to merge.

**Review by ryanloney — APPROVED** [review]

**Review by Debskij — APPROVED** [review]

**Pull request merged**